server/src/main/java/io/littlehorse/server/streams/topology/core/WfService.java

```diff
@@ -111,29 +111,20 @@ public PrincipalModel getPrincipal(PrincipalIdModel id) {
             id = new PrincipalIdModel(LHConstants.ANONYMOUS_PRINCIPAL);
         }
 
-        PrincipalModel storedResult = metadataManager.get(id);
-
-        if (storedResult == null && id.getId().equals(LHConstants.ANONYMOUS_PRINCIPAL)) {
-            // This means that all of the following are true:
-            // - The `anonymous` Principal has not yet been modified by the customer.
-            // - We just implicitly create it.
-            return createAnonymousPrincipal();
+        PrincipalModel principalModel = metadataManager.get(id);
+
+        if (principalModel == null && id.getId().equals(LHConstants.ANONYMOUS_PRINCIPAL)) {
+            // If Anonymous Principal missing from store (should never happen...)
+            // Implicitly create empty Anonymous Principal
```

Comment:
"Should Never Happen" is a code smell. Can you throw an `IllegalStateException`? Maybe @eduwercamacaro would have a better idea?

However, I disagree about "should never happen." What if we have 2 servers on version 0.11.2 in our cluster, and we start a rolling restart to get to 1.0.

If we first restart the server that _does not_ have the Metadata Partition assigned to it, then when it restarts, we'll have:
* 1.0 version server that doesn't own the Metadata Partition
* 0.11.2 version server that does own the Metadata Partition
* No `InitializationHistory` or `InitConfig` record created
* No `anonymous` Principal.

This tells me that we probably _do_ want to "create anonymous from thin air" in that case. But, as we said, it's a security vulnerability....what Kafka does is do _two_ rolling restarts. The first one is more lenient so "it works," then the second one changes a configuration so it would throw `IllegalStateException`.